pennylane/registers.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :func:`registers` function to build the registers for a given dictionary of registers
+"""
+
+from .wires import Wires
+
+
+def registers(register_dict, _start_wire_index=0):
+    """Returns a collection of wire registers when given a dictionary of register names and sizes
+    (number of qubits in the register).
+
+    The registers are a dictionary of :class:`~.Wires` objects where the key is the register name
+    and the value is the ``Wires`` object. The individual wire labels are integers, where the
+    ordering is based on appearance first, then on nestedness.
+
+    Args:
+        register_dict (dict): a dictionary of registers where the keys are the names (str) of the
+        registers and the values are either nested dictionaries of more registers or integers (int).
+        At the most nested level for each register key, the value must be an int.
```

Comment:
```suggestion
            registers and the values are either nested dictionaries of more registers or integers (int).
            At the most nested level for each register key, the value must be an int.
```
otherwise this won't render correctly in docs